tests/i18n/tests.py

```diff
@@ -1859,6 +1859,10 @@ def test_page_with_dash(self):
         response = self.client.get('/de-simple-page/')
         self.assertEqual(response.content, b'Yes')
 
+    def test_404_page_no_default_language_redirect(self):
+        response = self.client.get('/non-existent-no-slash', follow=False)
```

Comment:
On second thought, I'm actually in favour of expliciting follow=False, it prevents the test to pass if the redirection to the 404 page occurs.